test/binary_gray_test.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// binary_gray_test.dart
+// Tests for binary to gray and gray to binary conversion.
+//
+// 2023 October 5
+// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+import 'package:test/test.dart';
+
+Future<void> main() async {
```

Comment:
An interesting test might be to ensure that no sequential encodings have more than 1 bit difference and iterate through some?  Just to test the intent of the grey code regardless of the algorithm?